### PR TITLE
Np 2318 test for all types of events

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: search-commons/src/main/java/no/unit/nva/search/ElasticSearchHighLevelRestClient.java [hunk 1]
```diff
@@ -41,34 +41,27 @@
...
-
     public static final String INITIAL_LOG_MESSAGE = "using Elasticsearch endpoint {} and index {}";
     public static final String SOURCE_JSON_POINTER = "/_source";
     public static final String TOTAL_JSON_POINTER = "/hits/total/value";
     public static final String TOOK_JSON_POINTER = "/took";
     public static final String HITS_JSON_POINTER = "/hits/hits";
     public static final String DOCUMENT_WITH_ID_WAS_NOT_FOUND_IN_ELASTICSEARCH
             = "Document with id={} was not found in elasticsearch";
-
+    public static final URI DEFAULT_SEARCH_CONTEXT = URI.create("https://api.nva.unit.no/resources/search");
```
Comment (axthosarouris) [inline search-commons/src/main/java/no/unit/nva/search/ElasticSearchHighLevelRestClient.java:55]: automatic re-ordering

File: search-commons/src/main/java/no/unit/nva/search/ElasticSearchHighLevelRestClient.java [hunk 2]
```diff
@@ -117,6 +110,53 @@ public SearchResourcesResponse searchSingleTerm(String term,
         }
     }
 
+    /**
```
Comment (axthosarouris) [inline search-commons/src/main/java/no/unit/nva/search/ElasticSearchHighLevelRestClient.java:113]: changes due to automatic linting

File: search-commons/src/main/java/no/unit/nva/search/IndexDocument.java [hunk 3]
```diff
@@ -169,6 +168,26 @@ public Reference getReference() {
         return reference;
     }
 
+    @JacocoGenerated
```
Comment (axthosarouris) [inline search-commons/src/main/java/no/unit/nva/search/IndexDocument.java:171]: linting

File: search-commons/src/main/java/no/unit/nva/search/IndexDocument.java [hunk 4]
```diff
@@ -180,40 +199,20 @@ public boolean equals(Object o) {
...
-                owner,
-                description,
-                publicationAbstract,
-                publisher,
-                modifiedDate,
-                publishedDate,
-                alternativeTitles,
-                tags,
-                reference);
+               && Objects.equals(id, that.id)
```
Comment (axthosarouris) [inline search-commons/src/main/java/no/unit/nva/search/IndexDocument.java:202]: linting

File: search-commons/src/main/java/no/unit/nva/search/IndexDocument.java [hunk 5]
```diff
@@ -331,13 +329,13 @@ public Builder withReference(Reference reference) {
             return this;
         }
 
+        public IndexDocument build() {
```
Comment (axthosarouris) [inline search-commons/src/main/java/no/unit/nva/search/IndexDocument.java:332]: linting. Builder will be removed in future PR

File: search-resources-dynamodb-trigger/src/test/java/no/unit/nva/publication/TestDataGenerator.java [hunk 6]
```diff
@@ -55,8 +58,11 @@ public InputStream createResourceEvent(String eventType,
                                            PublicationStatus newPublicationStatus)
         throws JsonProcessingException, MalformedURLException, InvalidIssnException {
         initTemplate();
-
-        return generateModifyEvent(oldPublicationStatus, newPublicationStatus, eventType);
+        if (REMOVE.equals(eventType)) {
+            return generateRemoveEvent(oldPublicationStatus);
```
Comment (axthosarouris) [inline search-resources-dynamodb-trigger/src/test/java/no/unit/nva/publication/TestDataGenerator.java:62]: The action is decided on the content. If it contains a new image it is an update (insertion is treated as an update). If it contains only an old image it is a deletion.

**Review by brinxmat — APPROVED** [review]

**Pull request merged**